### PR TITLE
Implement Form Validation

### PR DESCRIPTION
**Pull request by mtahery20**
Added form validation with functionality to the following items:
-If user signs in with incorrect password or email address
-After user sends a password reset link, a success notification is shown

I also added a link to return back to the login page from the password reset page

### CODE REVIEW

**Review by ssarma107 — CHANGES_REQUESTED** [review]
everything is working except for the registration page, still needs form validation but approving for the log-in page and reset password page.

File: apps/web2/src/pages/FriendsPage/FriendsPage.tsx [hunk 1]
```diff
@@ -314,16 +340,36 @@ export default function FriendsPage(): JSX.Element {
...
+                )}
                 <form onSubmit={onAddFriendHandler}>
                   <label htmlFor="email-addres" className="sr-only">
                     Email address
                   </label>
                   <input
                     id="email-address"
                     name="email"
-                    type="email"
+                    type="text"
```
Comment (navarrojandg) [inline apps/web2/src/pages/FriendsPage/FriendsPage.tsx:371]: Believe that `type="email"` is correct here.

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/email

Comment (navarrojandg) [reply apps/web2/src/pages/FriendsPage/FriendsPage.tsx:371]: Making it `type="text"` allows us to accept any string input and rely on the regex test to validate the string as an email. Helps make errors uniform.

-- Marmar

**Comment by ivancuadra869** [comment]
The validation that is present works. There could be validation added, I'm going to approve this and merge. If you there is time, we'll take a look at the remaining validation

**Pull request merged**